core/includes/form.inc

```diff
@@ -1944,6 +1949,15 @@ function form_builder($form_id, &$element, &$form_state) {
       $element[$key]['#access'] = FALSE;
     }
 
+    // Count the elements at this level with IDs.
+    if (!isset($count_total)) {
+      $count_total = 0;
+    }
+    elseif (isset($element['#id'])) {
```

Comment:
How certain are we that all elements on the form will have an ID? Wouldn't it be better to just count the number of `<input>` elements...?

Reply:
All form API elements will have an ID since the ID is derived from the form API key. There are a lot of form elements that aren't `<input>` elements so I think using IDs will be more accurate (it won't always be perfect, but closer than leaving out selects, textareas, etc).